core/include/scipp/core/element/special_values.h

```diff
@@ -12,6 +12,58 @@
 
 namespace scipp::core::element {
 
+constexpr auto isnan =
+    overloaded{arg_list<double, float>,
+               [](const auto x) {
+                 using std::isnan;
+                 return isnan(x);
+               },
+               [](const units::Unit &) { return units::dimensionless; }};
+
+constexpr auto isinf =
+    overloaded{arg_list<double, float>,
+               [](const auto x) {
+                 using std::isinf;
+                 return isinf(x);
+               },
+               [](const units::Unit &) { return units::dimensionless; }};
+
+constexpr auto isfinite =
+    overloaded{arg_list<double, float>,
+               [](const auto x) {
+                 using std::isfinite;
+                 return isfinite(x);
+               },
+               [](const units::Unit &) { return units::dimensionless; }};
+
+namespace detail {
+template <typename T>
+auto isposinf(T x) -> std::enable_if_t<std::is_floating_point_v<T>, bool> {
+  return std::isinf(x) && !std::signbit(x);
+}
+
+template <typename T>
+auto isneginf(T x) -> std::enable_if_t<std::is_floating_point_v<T>, bool> {
+  return std::isinf(x) && std::signbit(x);
+}
+} // namespace detail
+
+constexpr auto isposinf =
+    overloaded{arg_list<double, float>,
+               [](const auto x) {
+                 using detail::isposinf;
+                 return isposinf(x);
```

Comment:
Why the helper function in `detail`? Is it used in multiple places?

Reply:
No.
See the reply on SFINEA above. Not sure if there is a way to select an implementation for floats or `ValueAndVariance` purely inside of the kernel. Do you have a better suggestion?

Reply:
No better suggestion, I think. I suppose another way would have been to use
```cpp
using std::isinf;
using std::signbit;
return isinf(x) && !signbit(x);
```
so ADL works? Not sure if that is better.

Reply:
That would mean implementing `signbit` for `ValueAndVariance`. I'm already not happy with how `isinf` is implemented for that and I think `signbit` would be even more confusing / ambiguous for a caller.

Reply:
I'm ok with leaving it.

If you have a better idea on how to implement `isinf`, suggestions are welcome!

Reply:
It's a question of what is needed. Do people need to filter their data based on the values alone or does it make sense to also filter variances? Can variances even be infinite or NaN when the value isn't?

Reply:
Good question, we discussed this ambiguity before (also in the context of comparison operators) and did not find a satisfying answer.

I suppose there may be cases where the error propagation divides by zero so you may be inf variances, even if the value is not? Not sure that can happen for any of our current operations though?